otx/algorithms/action/adapters/mmaction/utils/config_utils.py

```diff
@@ -0,0 +1,108 @@
+"""Collection of utils for task implementation in Action Task."""
+
+# Copyright (C) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import os
+from collections import defaultdict
+from typing import List, Union
+
+from mmcv.utils import Config, ConfigDict
+
+from otx.algorithms.common.adapters.mmcv.utils import get_data_cfg, prepare_work_dir
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.label import LabelEntity
+from otx.api.usecases.reporting.time_monitor_callback import TimeMonitorCallback
+from otx.api.utils.argument_checks import (
+    DatasetParamTypeCheck,
+    check_input_parameters_type,
+)
+
+
+@check_input_parameters_type()
+def patch_config(config: Config, base_dir: str, work_dir: str):
+    """Patch recipe config suitable to mmaction."""
+    # FIXME omnisource is hard coded
+    config.omnisource = None
+    config.work_dir = work_dir
+    patch_data_pipeline(config, base_dir)
+    patch_datasets(config)
+
+
+@check_input_parameters_type()
+def patch_data_pipeline(config: Config, base_dir: str):
```

Comment:
This function is also used in detection. How about moving it to `common`?

Reply:
Done. Detection tasks also should be changed, but it is not this PR' range so I just add TODO flag on it